File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        
        int left = 0;
        int maxCount = 0;
        int zeroCount = 0;
        
        for(int r = 0; r < nums.size(); r++) {
            int curInt = nums[r];
            
            if(curInt == 0) {
                zeroCount++;
            }
            
            while(zeroCount > k) {
                if(nums[left] == 0) {
                    zeroCount--;
                }
                left++;
            }
            
            maxCount = max(maxCount, r - left + 1);
        }
        
        return maxCount;
        
    }
};

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        
        left = 0
        maxCount = 0
        zeroCount = 0
        
        for i in range(len(nums)):
            if nums[i] == 0:
                zeroCount+=1;
                
            while zeroCount > k:
                if nums[left] == 0:
                    zeroCount-=1;
                left+=1
        
            maxCount = max(maxCount, i - left + 1)
            
            
        return maxCount
        